File: DatingApp/API/Helpers/UserParams.cs
namespace API.Helpers;

public class UserParams : PaginationParams
{
    public string? Gender { get; set; }
    public string? CurrentUsername { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 60;
    public List<string>? Interests { get; set; }
    public string OrderBy { get; set; } = "lastActive";
}

File: DatingApp/client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
  gender: string;
  orderBy = 'lastActive';
  minAge = 18;
  maxAge = 65;
  interests: string[] = [];
  currentPageNumber = 1;
  itemsPerPage = 6;

  constructor(user: User | null) {
    this.gender = user?.gender === 'female' ? 'male' : 'female';
  }
}
